File: tests/RoslynDiff.Core.Tests/TestFixtures/VisualBasic/ClassChanges/ClassModified_New.vb
' Test fixture: Class modification detection
' Expected: Class "Person" should be detected as Modified
Namespace TestFixtures
    ''' <summary>
    ''' A person class for testing class modification.
    ''' </summary>
    Public Class Person
        Private _name As String
        Private _age As Integer

        ''' <summary>
        ''' Creates a new Person instance.
        ''' </summary>
        Public Sub New(name As String, age As Integer)
            _name = name
            _age = age
        End Sub

        ''' <summary>
        ''' Gets the person's name.
        ''' </summary>
        Public Function GetName() As String
            Return _name
        End Function

        ''' <summary>
        ''' Gets the person's age.
        ''' </summary>
        Public Function GetAge() As Integer
            Return _age
        End Function
    End Class
End Namespace

File: tests/RoslynDiff.Core.Tests/TestFixtures/CSharp/ClassChanges/ClassModified_New.cs
// Test fixture: Class modification detection
// Expected: Class "DataProcessor" should be detected as Modified
//           (base class added, interface implemented)
namespace TestFixtures;

/// <summary>
/// A data processor class with modified implementation.
/// Now inherits from BaseProcessor and implements IProcessor.
/// </summary>
public class DataProcessor : BaseProcessor, IProcessor
{
    private readonly string _name;
    private readonly ILogger _logger;

    public DataProcessor(string name, ILogger logger)
    {
        _name = name;
        _logger = logger;
    }

    public override string Process(string input)
    {
        _logger.Log($"Processing: {input}");
        return input.ToUpper().Trim();
    }
}

public abstract class BaseProcessor
{
    public abstract string Process(string input);
}

public interface IProcessor
{
    string Process(string input);
}

public interface ILogger
{
    void Log(string message);
}
